ui/src/router/versions/list/ListRouterVersionsTable.js

```diff
@@ -22,15 +21,70 @@ export const ListRouterVersionsTable = ({
   error,
   deployedVersion,
   renderActions,
-  onRowClick
+  onRowClick,
+  ...props
 }) => {
-  const cellProps = item =>
-    onRowClick
+  const [selection, setSelection] = useState({});
+
+  const isSelected = useCallback(item => !!selection[item.version], [
+    selection
+  ]);
+
+  const readyForDiff = useMemo(() => Object.entries(selection).length === 2, [
+    selection
+  ]);
+
+  const selectForComparison = useCallback(
+    item => {
+      setSelection(items => ({
+        ...items,
+        [item.version]: item
+      }));
+    },
+    [setSelection]
+  );
+
+  const unselectFromComparison = useCallback(
+    item => {
+      setSelection(items => {
+        delete items[item.version];
+        return { ...items };
+      });
+    },
+    [setSelection]
+  );
+
+  const onShowDiff = useCallback(() => {
+    const selected = Object.values(selection);
+    if (selected.length === 2) {
+      const [leftSelection, rightSelection] = selected;
```

Comment:
I thought that it's fine to have it this way for simplicity, but since you brought it up, I updated the code to show versions in the order they were selected in the table